### ISSUE TITLE
long float fields incomment line break gap_fit

### ISSUE THREAD

**Issue by bernstei** [libAtoms/QUIP#561]
It's not uncommon for the workflow to have very long float vectors in the Atoms info (comment) line, and that breaks `gap_fit` because it's either too long (easy to fix) or something other than 1, 3, 6, or 9 fields (harder to fix).    I'm not sure if that'd take switching to the new xyz reader @jameskermode , but it'd be nice to deal with this.

**Comment by bernstei** [comment]
Not a big deal, by the way, to just filter those out in the workflow.  But it's an unnecessary limitation on gap_fit.

**Comment by bernstei** [comment]
OK, I fixed the random errors, but the test fails on the actual reference quantities.  I suggest running it locally with `pytest` rather than relying on the CI, but I tried to give a more informative error.  I'm not sure if the summary outputted by `pytest` is sufficient, but if you run `pytest -s` and scroll back you can find what actually failed.  In my case it's a quantitative mismatch in the error table, specifically
```
E   AssertionError: Mismatch for ref prop atomization_energy/atom config train measure RMSE ref 0.014974721709894103 != actual 0.01606044820056456
```

FWIW, it's not deterministic - I think I've seen a range of values for the actual error from 0.016 - 0.020 so far.

**Comment by jameskermode** [comment]
Last comment doesn’t seem to follow- does it belong elsewhere?

Agree it would be good to fix this. If you could confirm that the new extxyz parser doesn’t choke on your long comment lines that would be useful and would increase the motivation to make the change in QUIP/GAP.

**Comment by bernstei** [comment]
> Last comment doesn’t seem to follow- does it belong elsewhere?

Yeah, oops.
> 
> Agree it would be good to fix this. If you could confirm that the new extxyz parser doesn’t choke on your long comment lines that would be useful and would increase the motivation to make the change in QUIP/GAP.

Is there a branch for that?  If we've considering switching, I think we should make a PR.

**Comment by bernstei** [comment]
I just checked with the `extxyz` executable in the python package.  Once I quoted a string that contained `=`, it claimed to read the header without an error.  When I run `extxyz.read()` myself, it complains about the long header line, but if I shorten it just a bit it has no problems with the large number of float components in the SOAP descriptor field.  Is there a max line length, of order 2048, like in the xyz.c currently in libQUIP?

**Comment by bernstei** [comment]
Looks like a bug in the reader that extends the line length as needed (`libextxyz/extxyz.c`).  It extends once if needed, but not twice.  I'll try to debug it.

**Comment by bernstei** [comment]
new extxyz reader works after bugfix https://github.com/libAtoms/extxyz/pull/10

**Comment by bernstei** [comment]
How do we feel about the fact that the new reader isn't 100% backward compatible with the old one, e.g. requiring quotes for strings with `=`?  I suspect those are rare, but there will be issues.

**Comment by jameskermode** [comment]
Agreed. We should setup a PR in QUIP+GAP for this and test for a while before merging. I'll try to do that soon.

**Comment by jameskermode** [comment]
@Sideboard is also going to discuss the pathway to ASE adopting extxyz as an optional dependency when he attends the ASE hackathon in a couple of weeks' time.
